packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewReconnectionToolsExecutor.java

```diff
@@ -79,4 +138,37 @@ private VariableManager createVariableManager(ReconnectionToolInterpreterData to
         return variableManager;
     }
 
+    private List<EPackage> getAccessibleEPackages(IEditingContext editingContext) {
+        if (editingContext instanceof EditingContext) {
+            Registry packageRegistry = ((EditingContext) editingContext).getDomain().getResourceSet().getPackageRegistry();
+            // @formatter:off
+            return packageRegistry.values().stream()
+                                  .filter(EPackage.class::isInstance)
+                                  .map(EPackage.class::cast)
+                                  .collect(Collectors.toList());
+            // @formatter:on
+        } else {
+            return List.of();
+        }
+    }
+
+    private AQLInterpreter createInterpreter(View view, List<EPackage> visibleEPackages) {
+        AutowireCapableBeanFactory beanFactory = this.applicationContext.getAutowireCapableBeanFactory();
+        // @formatter:off
+        List<Object> serviceInstances = this.javaServiceProviders.stream()
+                .flatMap(provider -> provider.getServiceClasses(view).stream())
+                .map(serviceClass -> {
+                    try {
+                        return beanFactory.createBean(serviceClass);
+                    } catch (BeansException beansException) {
+                        this.logger.warn("Error while trying to instantiate Java service class " + serviceClass.getName(), beansException); //$NON-NLS-1$
+                        return null;
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        // @formatter:on
+        return new AQLInterpreter(List.of(), serviceInstances, visibleEPackages);
+    }
```

Comment:
This is a recurring need, and these lines are duplicated in several places (not always exactly identical, sometimes for good reasons sometimes because they evolved independently).
I'm not sure which mecahnism would be the best, but we should have a way to share these instances in the relevant scopes.

Reply:
I think that pretty much every location where these pieces of code are currently used will be re-written or heavily impacted when we will start working on [this issue](https://github.com/eclipse-sirius/sirius-components/issues/1422).

The code that you have highlighted and a bit more above too will have to be thrown away soon.

Reply:
FYI I had to do the same in the "AQL completion" PR as I need an interpreter to answer the completion query, and this is in a context outside of a diagram rendering or "classical" tool application.
There is also one instance in `ViewInitialDirectEditElementLabelProvider.createInterpreter(View, IEditingContext)` as it needs the evaluate AQL expressions tool.